plugins/Network/plugin.py

```diff
@@ -165,7 +165,24 @@ def cmdall(self, irc, msg, args, commandAndArgs):
         for ircd in ircs:
             self.Proxy(ircd, msg, commandAndArgs)
     cmdall = wrap(cmdall, ['admin', many('something')])
-    
+
+    def asrvchanscmd(self, irc, msg, args, commandAndArgs):
+        """<command> <args>...
+        Perform <command> (with associated <arg>s) on all networks and all channels."""
+        ircs = world.ircs
+        for ircd in ircs:
+            for channel in ircd.state.channels.keys():
+                self.Proxy(ircd, msg, commandAndArgs)
+    asrvchanscmd = wrap(asrvchanscmd, ['admin', many('something')])
+
+    def achanscmd(self, irc, msg, args, commandAndArgs):
+        """<command> <args>...
+        Perform <command> (with associated <arg>s on all channels on current network."""
+        ircd = world.getIrc(irc.network)
+        for channel in irc.state.channels.keys():
+            self.Proxy(ircd, msg, commandAndArgs)
```

Comment:
This is misleading because doesn't actually change the target channel that the command is being run in. Suppose you're in `#test` and run an echo:

```
06:51 <~GLolol> `achanscmd echo $channel
06:51 <@Lily> ​#test
06:51 <@Lily> ​#test
06:51 <@Lily> ​#test
06:51 <@Lily> ​#test
06:51 <@Lily> ​#test
06:51 <@Lily> ​#test
06:51 <@Lily> ​#test
06:51 <@Lily> ​#test
```

All that happens is that the same command runs multiple times, once for each channel the bot is in. Maybe you could try overriding `msg.args[0]`?